### DISCUSSION TITLE
Show theia only inside IFrame

### DISCUSSION THREAD

**Discussion by ll9** [eclipse-theia/theia#9217] [Q&A]
I am running theia as a docker container and embed it into my own website as an IFrame. My current setup looks like this:

dockerUrl: http://localhost:8080
Website Url: http://localhost:5000 and http://localhost:5000/theia (embeds http://localhost:8080 as an IFrame).

I want theia to be available on http://localhost:5000/theia as an embedded IFrame only. I do not want users to be able to visit the docker container directly on http://localhost:5000.

How can I do that?

**Comment by akosyakov** [comment]
You will need a proxy to deploy theia on context path or some custom extension which use express.js middleware to change the base path.

**Comment by ll9** [comment]
Could you elaborate a bit more? Unfortunately I don't really understand how changing the base path in express would help me hide theia if it isn't in an IFrame.

**Comment by talsenteam** [comment]
> I am running theia as a docker container and embed it into my own website as an IFrame. My current setup looks like this:
> 
> dockerUrl: http://localhost:8080
> Website Url: http://localhost:5000 and http://localhost:5000/theia (embeds http://localhost:8080 as an IFrame).
> 
> I want theia to be available on http://localhost:5000/theia as an embedded IFrame only. I do not want users to be able to visit the docker container directly on http://localhost:5000.
> 
> How can I do that?

FYI: If you trying to prevent direct access to theia, a simple IFrame solution will not be sufficient. Because the page does not know if it is loaded via IFrame or not. A (e.g.) JWT based request for the editor would be required (what I found out until now, correct me if you find a simpler / better approach).
So something like the following:
1. Front-end loads and receives JWT from server (protect unwanted access via LogIn)
2. Front-end requests theia editor using its JWT
3. Theia delivery server checks the JWT and delivers only when JWT is valid
4. Front-end embeds theia IFrame into page
